chainer/functions/array/hstack.py

```diff
@@ -53,11 +53,31 @@ def hstack(xs):
     """Concatenate variables horizontally (column wise).
 
     Args:
-        xs (list of chainer.Variable): Variables to be concatenated.
+        xs (list of :class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variables to be concatenated. The variables must have the
+            same shape, except in the horizontal (column wise) dimension.
```

Comment:
It seems `F.stack` does not raise `InvalidType` error when array is NumPy and `ndim` is 0. Could you check it does work other cases?

```
In [1]: import numpy as np
In [2]: from chainer import functions as F
In [3]: a = np.array(4)
In [4]: b = np.array(4)
In [5]: F.stack((a, b))
Out[5]: <variable at 0x113acd860>
In [6]: F.stack((a, b)).data
Out[6]: array([4, 4])
```